cmd/opentelemetry-collector/app/defaults/default_config.go

```diff
@@ -83,9 +93,15 @@ func createReceivers(factories config.Factories) configmodels.Receivers {
 			},
 		},
 	}
-	return map[string]configmodels.Receiver{
-		"jaeger": rec,
+	recvs := map[string]configmodels.Receiver{
+		"jaeger": jaeger,
+	}
+	if zipkinHostPort != disabledZipkinHostPort {
```

Comment:
Rather than define a special port, would it be better to just have a pointer? So test against nil.

Reply:
We could do that too, although this uses the same approach as other parts of the codebase - zipkin is disabled if `:0` is used. This value is also set as the default of the flag.

Reply:
I did a small change, I have removed the const and used a built-in func to return the value of disabled endpoint.